airbyte/results.py

```diff
@@ -12,17 +12,34 @@
 
     from sqlalchemy.engine import Engine
 
+    from airbyte._future_cdk.catalog_providers import CatalogProvider
+    from airbyte._future_cdk.state_writers import StateWriterBase
     from airbyte.caches import CacheBase
+    from airbyte.destinations.base import Destination
+    from airbyte.progress import ProgressTracker
+    from airbyte.sources.base import Source
 
 
 class ReadResult(Mapping[str, CachedDataset]):
+    """The result of a read operation.
+
+    This class is used to return information about the read operation, such as the number of
+    records read. It should not be created directly, but instead returned by the write method
+    of a destination.
+    """
+
     def __init__(
         self,
+        *,
+        source_name: str,
         processed_records: int,
```

Comment:
The `process_records` arg should be replaced by the input from `progress_tracker`.